#!/usr/bin/env python
PACKAGE = "auv_mapping"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Docking approach target offsets (relative to docking_station frame)
gen.add("approach_offset_x", double_t, 0, "Approach target X offset (m)", 0.0, -1.0, 1.0)
gen.add("approach_offset_y", double_t, 0, "Approach target Y offset (m)", 0.0, -1.0, 1.0)
gen.add("approach_offset_z", double_t, 0, "Approach target Z offset (m) above station", 1.0, 0.1, 3.0)

# Docking puck target offsets (final docking position)
gen.add("puck_offset_x", double_t, 0, "Puck target X offset (m)", 0.0, -1.0, 1.0)
gen.add("puck_offset_y", double_t, 0, "Puck target Y offset (m)", 0.0, -1.0, 1.0)
gen.add("puck_offset_z", double_t, 0, "Puck target Z offset (m) above station", 0.5, 0.0, 2.0)

exit(gen.generate(PACKAGE, "auv_mapping", "DockingTrajectory"))
